inference-engine/src/mkldnn_plugin/nodes/mkldnn_quantize_node.cpp

```diff
@@ -136,6 +136,23 @@ void MKLDNNQuantizeNode::init() {
 
     bool binarization = levels == 2;
 
+    auto isCorrectRanges = [](bool low, bool high, const float *lowDataPtr, const float *highDataPtr, const size_t size) {
+        for (size_t i = 0; i < size; i++) {
+            float lowData = lowDataPtr[low ? 0 : i];
+            float highData = highDataPtr[high ? 0 : i];
+            if (highData == lowData)
+                return false;
+        }
+        return true;
+    };
+
+    if ((!isCorrectRanges(isInputLowBroadcasted, isInputHighBroadcasted, inputLowData, inputHighData,
+        std::max(inputLowBlob->size(), inputHighBlob->size())) && !binarization) ||
+        !isCorrectRanges(isOutputLowBroadcasted, isOutputHighBroadcasted, outputLowData, outputHighData,
+        std::max(outputLowBlob->size(), outputHighBlob->size()))) {
+            THROW_IE_EXCEPTION << "Quantize layer with name '" << getName() << "' has wrong quantize ranges";
```

Comment:
Can we integrate this check inside cycles on 237 and 245 lines, so to avoid one more pass through values?